BubblesKotlin/app/src/main/java/com/example/android/bubbles/data/NotificationHelper.kt

```diff
@@ -61,16 +67,45 @@ class NotificationHelper(private val context: Context) {
                 }
             )
         }
+        updateShortcuts()
     }
 
     @WorkerThread
-    fun showNotification(chat: Chat, fromUser: Boolean) {
-        val icon = Icon.createWithAdaptiveBitmap(
-            BitmapFactory.decodeResource(
-                context.resources,
-                chat.contact.icon
+    fun updateShortcuts() {
+        val shortcuts = Contact.CONTACTS.map { contact ->
+            val icon = Icon.createWithAdaptiveBitmap(
+                context.resources.assets.open(contact.icon).use { input ->
+                    BitmapFactory.decodeStream(input)
+                }
             )
-        )
+            ShortcutInfo.Builder(context, contact.id.toString())
+                .setActivity(ComponentName(context, MainActivity::class.java))
+                .setShortLabel(contact.name)
+                .setIcon(icon)
+                .setIntent(
+                    Intent(context, MainActivity::class.java)
+                        .setAction(Intent.ACTION_VIEW)
+                        .setData(
+                            Uri.parse(
+                                "https://android.example.com/chat/${contact.id}"
+                            )
+                        )
+                )
+                .setPerson(
+                    Person.Builder()
+                        .setName(contact.name)
+                        .setIcon(icon)
+                        .build()
+                )
+                .build()
+        }
+        shortcutManager.addDynamicShortcuts(shortcuts)
+    }
+
+    @WorkerThread
+    fun showNotification(chat: Chat, fromUser: Boolean) {
+        Icon.createWithAdaptiveBitmapContentUri("content://")
```

Comment:
Empty uri?

Reply:
Removed. This entire line was totally useless.